system/Validation/FormatRules.php

```diff
@@ -346,4 +346,33 @@ public function valid_date(?string $str = null, ?string $format = null): bool
 
         return $date !== false && $errors !== false && $errors['warning_count'] === 0 && $errors['error_count'] === 0;
     }
+
+    /**
+     * Token JWT Validator
+     *
+     * @param string $str String
+     */
+    public function valid_jwt(?string $str = null): bool
+    {
+        if (empty($str)) {
+            return false;
+        }
+
+        $segments = explode('.', $str);
+
+        if (count($segments) !== 3 || in_array('', $segments, true)) {
+            return false;
+        }
+
+        foreach ($segments as $index => $value) {
+            if ($index < 2) {
+                $encoded = rtrim(base64_encode(base64_decode($value, true)), '==');
+                if ($encoded !== $value) {
+                    return false;
+                }
+            }
+        }
+
+        return mb_strlen($segments[$index], '8bit') >= 32 && mb_strlen($segments[$index], '8bit') <= 64;
```

Comment:
Why is the length 32 to 64?

Reply:
In algo SHA256-SHA512 & Sodium, get min 32 and max 64 byte.

Ref : [firebase/php-jwt](https://github.com/firebase/php-jwt/blob/main/src/JWT.php#L55) & [JWT.io](https://jwt.io/libraries?language=PHP)

Reply:
Is this not valid?
```
eyJhbGciOiJSUzUxMiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImlhdCI6MTUxNjIzOTAyMn0.jYW04zLDHfR1v7xdrW3lCGZrMIsVe0vWCfVkN2DRns2c3MN-mcp_-RE6TN9umSBYoNV-mnb31wFf8iun3fB6aDS6m_OXAiURVEKrPFNGlR38JSHUtsFzqTOj-wFrJZN4RwvZnNGSMvK3wzzUriZqmiNLsG8lktlEn6KA4kYVaM61_NpmPHWAjGExWv7cjHYupcjMSmR8uMTwN5UuAwgW6FRstCJEfoxwb0WKiyoaSlDuIiHZJ0cyGhhEmmAPiCwtPAwGeaL1yZMcp0p82cpTQ5Qb-7CtRov3N4DcOHgWYk6LomPR5j5cCkePAz87duqyzSMpCB0mCOuE3CU2VMtGeQ
```

Reply:
Ohh sorry, already added length for RS, ES and PS

Sample :
`
$str = array(
    'jYW04zLDHfR1v7xdrW3lCGZrMIsVe0vWCfVkN2DRns2c3MN-mcp_-RE6TN9umSBYoNV-mnb31wFf8iun3fB6aDS6m_OXAiURVEKrPFNGlR38JSHUtsFzqTOj-wFrJZN4RwvZnNGSMvK3wzzUriZqmiNLsG8lktlEn6KA4kYVaM61_NpmPHWAjGExWv7cjHYupcjMSmR8uMTwN5UuAwgW6FRstCJEfoxwb0WKiyoaSlDuIiHZJ0cyGhhEmmAPiCwtPAwGeaL1yZMcp0p82cpTQ5Qb-7CtRov3N4DcOHgWYk6LomPR5j5cCkePAz87duqyzSMpCB0mCOuE3CU2VMtGeQ',
    'SefW4LRE8rlUTjvGXEuC5oqqW0jsjaUR9pBnTt9Jarp3EjDo3v_djd8nN9Cwa_eFOvxkF61PERv08DiIb7DU1ceGOVhh49btyjAe5hxqo_fKS0c95PNV4ANgv9ca3ZZ1Df2i3dSRRmo-9yDg54QIgWFrOGu8t2bgbZfVnOv_qSJdagMa9QxhgAqRJd9Jrz0qEMLYPuLdIn-5XHYk07pruFhHgjcycGXYjwYSfH0kOke2K6NsX3VAjsHgU5GLVU9J9XCnyP5lx2cgk3IWF9y03fykigh820hqHWs28dwv-BHcMscumOvlAGW6YrnQqjreNtaLIwqs7LdjlHnqc5ybIQ',
    'Sv4y1O38_Pj9JMpWz_dn7Ms1LJBVGw-TId6a0lwTn0coTXo_JAbjIAogiMya8mvhF0nIkt7lF0WHSgL5kBuXtODrwPHbQPOtLeumkJ2N2sRDsN0LP0m_0QN7gCEkFSZ8FXukaPM8NqJsoNXEpeNLPzDoACNNtRZj5JiNwnxtpQt1CMunzUSeo_Cul7wHojlMy-Y5duxXnWTt_4rVVTspUjWcxBQxdRM1QQ_mP3eHN2ABVcBdkrRDK-szZn5RoHA6nPfKTZ51QX3UPT5eGJb1MsQGSKC2vlSi-TPP8snXCU7SqdyXC6rchLXtH_pKWwe4vDFwOcxhIgP3XzKTEO7nAw',
    'QrOYZgxP7_DfAuMnppKVPPtCMrxVIwJFPUe47-Cnw_of50pEoZ5M57utxS78FAq48UnYyu76nMe2CKfjhj0LvI3WIbhTnrxdrs6tQiJZHs-j2qwcHRCfSFlqKN1HuOWDAXnsQd1CzKa597urll3MxdeMIN3PKFaWCUVozlrodr_qqKDOxeCpl64mPfCsKb22U_SYKn5kQLg9qPjPGayXnCF6iZiOAahN9PqzjWIjEe0lkJTySS9KJUPpSKhVjRuwabELsW4DDtp6Q9KCVZQUw5VwJVWm9vqhRUTfbPoC9WViOIYvBnjOf_eYMFm1F2y7Ee2dnj4XBI7gg8_tTX-SJQ',
    'ufxA9G1842f7K5tmDrbGh2gljMI9wNcspQjei0U7xipJwGBopUlc7HC-83IMEcnPLluOfB11GnJIdXUa-JC9VQ1klHvjMJ4t0rQKyPqd_BgJ5beuyjBmpCXkE5uR-4KolLZKlwCzPll8TzpRvMSQiamx1c6MqQmP2Vy-gOGfi-LTNNZhcr4lmyEVc73okFZAAB0nbnZ-GLN7qRc0ZtBLjOdR-R60yhihCw_nEKP3auvKnzi7hC_DwdT8bxkbcHAEE4AeyvERUm6dleoMqax55qLO2sHQ8ZnnsP8eUFDGvDn--rLyNIa0jklNW5LWLVIF3gmmpTOyQ2wldElJqmllhQ',
    'W7Sb3oA56Cp_sdRLQ7UzUiPlGqOZr24UVzY7ZB84MuZHM-TnDgi9QJTE5fX87JM6Q-_TyDjI5DY0l1CCoHIohbjPJTZAhmfyJfbeMsQH7nZv0hGlzRIUA_5bpLjHQPQk_X5J-_9RvCe5wFUq9d1h3Uvz_UTNLoF2j26nENFdDJ2JTsJ0OH3ihsQ5LYPAcsNtTET0gFSJWfBClqfrFbPcGe1Y54sMvBuf8fNMhhemPDIP3HGTpbSY-OmQ-jcvLp30ld38LQEklBReI5cS5J9vpST0fgkZ02v-Ex-s1Yt6TErldBOpjEAhVCMTciFLepp1kzyNO0AhlfWXl7QoEsSfWQ',
    'CUdWRfjVbbDnZNLyuJjgiEr85ngGZ4DXjKDhGCKX5pH2NVkNwIwM1y9zw9FN2MiSFG8tzWdjCBxkVSmGzqhFCkz65Gx5OffnD57iHy0cc56bgGfNq-WHlqvAUiAJX3HzJ5MRn2HDZ4eGm0bmBQDhB91qvDpNIyIaLHgxWl8Brmn-tLuxIt4B1kpKKFy-crCeAsdHkZABFvGmtNeelH_XpJWc-5DKTd6JtpIhpPrsUKd8iFvWYJkc9Vp1k3OXTZARzuJ7qkt2xCBC38wKU4U_KXYmsJ5fAUuXWwk4AoirYlGalLLXIppg3T_Lg8AGKrJpM1tXD-GbZyK4tL1M2O2lSg',
    'jYW04zLDHfR1v7xdrW3lCGZrMIsVe0vWCfVkN2DRns2c3MN-mcp_-RE6TN9umSBYoNV-mnb31wFf8iun3fB6aDS6m_OXAiURVEKrPFNGlR38JSHUtsFzqTOj-wFrJZN4RwvZnNGSMvK3wzzUriZqmiNLsG8lktlEn6KA4kYVaM61_NpmPHWAjGExWv7cjHYupcjMSmR8uMTwN5UuAwgW6FRstCJEfoxwb0WKiyoaSlDuIiHZJ0cyGhhEmmAPiCwtPAwGeaL1yZMcp0p82cpTQ5Qb-7CtRov3N4DcOHgWYk6LomPR5j5cCkePAz87duqyzSMpCB0mCOuE3CU2VMtGeQ', // Sample kenjis
    'tyh-VfuzIxCyGYDlkBA7DfyjrqmSHu6pQ2hoZuFqUSLPNY2N0mpHb3nk5K17HWP_3cYHBw7AhHale5wky6-sVA', // ES256
    'VUPWQZuClnkFbaEKCsPy7CZVMh5wxbCSpaAWFLpnTe9J0--PzHNeTFNXCrVHysAa3eFbuzD8_bLSsgTKC8SzHxRVSj5eN86vBPo_1fNfE7SHTYhWowjY4E_wuiC13yoj', // ES384
    'AbVUinMiT3J_03je8WTOIl-VdggzvoFgnOsdouAs-DLOtQzau9valrq-S6pETyi9Q18HH-EuwX49Q7m3KC0GuNBJAc9Tksulgsdq8GqwIqZqDKmG7hNmDzaQG1Dpdezn2qzv-otf3ZZe-qNOXUMRImGekfQFIuH_MjD2e8RZyww6lbZk', // ES512
    'iOeNU4dAFFeBwNj6qdhdvm-IvDQrTa6R22lQVJVuWJxorJfeQww5Nwsra0PjaOYhAMj9jNMO5YLmud8U7iQ5gJK2zYyepeSuXhfSi8yjFZfRiSkelqSkU19I-Ja8aQBDbqXf2SAWA8mHF8VS3F08rgEaLCyv98fLLH4vSvsJGf6ueZSLKDVXz24rZRXGWtYYk_OYYTVgR1cg0BLCsuCvqZvHleImJKiWmtS0-CymMO4MMjCy_FIl6I56NqLE9C87tUVpo1mT-kbg5cHDD8I7MjCW5Iii5dethB4Vid3mZ6emKjVYgXrtkOQ-JyGMh6fnQxEFN1ft33GX2eRHluK9eg', // PS256
);

foreach ($str as $key => $val){
    echo mb_strlen($val).PHP_EOL;
}
`

Reply:
Please use ` ```php ` and ` ``` ` for code block.

Reply:
```
$str = array(
    'jYW04zLDHfR1v7xdrW3lCGZrMIsVe0vWCfVkN2DRns2c3MN-mcp_-RE6TN9umSBYoNV-mnb31wFf8iun3fB6aDS6m_OXAiURVEKrPFNGlR38JSHUtsFzqTOj-wFrJZN4RwvZnNGSMvK3wzzUriZqmiNLsG8lktlEn6KA4kYVaM61_NpmPHWAjGExWv7cjHYupcjMSmR8uMTwN5UuAwgW6FRstCJEfoxwb0WKiyoaSlDuIiHZJ0cyGhhEmmAPiCwtPAwGeaL1yZMcp0p82cpTQ5Qb-7CtRov3N4DcOHgWYk6LomPR5j5cCkePAz87duqyzSMpCB0mCOuE3CU2VMtGeQ',
    'SefW4LRE8rlUTjvGXEuC5oqqW0jsjaUR9pBnTt9Jarp3EjDo3v_djd8nN9Cwa_eFOvxkF61PERv08DiIb7DU1ceGOVhh49btyjAe5hxqo_fKS0c95PNV4ANgv9ca3ZZ1Df2i3dSRRmo-9yDg54QIgWFrOGu8t2bgbZfVnOv_qSJdagMa9QxhgAqRJd9Jrz0qEMLYPuLdIn-5XHYk07pruFhHgjcycGXYjwYSfH0kOke2K6NsX3VAjsHgU5GLVU9J9XCnyP5lx2cgk3IWF9y03fykigh820hqHWs28dwv-BHcMscumOvlAGW6YrnQqjreNtaLIwqs7LdjlHnqc5ybIQ',
    'Sv4y1O38_Pj9JMpWz_dn7Ms1LJBVGw-TId6a0lwTn0coTXo_JAbjIAogiMya8mvhF0nIkt7lF0WHSgL5kBuXtODrwPHbQPOtLeumkJ2N2sRDsN0LP0m_0QN7gCEkFSZ8FXukaPM8NqJsoNXEpeNLPzDoACNNtRZj5JiNwnxtpQt1CMunzUSeo_Cul7wHojlMy-Y5duxXnWTt_4rVVTspUjWcxBQxdRM1QQ_mP3eHN2ABVcBdkrRDK-szZn5RoHA6nPfKTZ51QX3UPT5eGJb1MsQGSKC2vlSi-TPP8snXCU7SqdyXC6rchLXtH_pKWwe4vDFwOcxhIgP3XzKTEO7nAw',
    'QrOYZgxP7_DfAuMnppKVPPtCMrxVIwJFPUe47-Cnw_of50pEoZ5M57utxS78FAq48UnYyu76nMe2CKfjhj0LvI3WIbhTnrxdrs6tQiJZHs-j2qwcHRCfSFlqKN1HuOWDAXnsQd1CzKa597urll3MxdeMIN3PKFaWCUVozlrodr_qqKDOxeCpl64mPfCsKb22U_SYKn5kQLg9qPjPGayXnCF6iZiOAahN9PqzjWIjEe0lkJTySS9KJUPpSKhVjRuwabELsW4DDtp6Q9KCVZQUw5VwJVWm9vqhRUTfbPoC9WViOIYvBnjOf_eYMFm1F2y7Ee2dnj4XBI7gg8_tTX-SJQ',
    'ufxA9G1842f7K5tmDrbGh2gljMI9wNcspQjei0U7xipJwGBopUlc7HC-83IMEcnPLluOfB11GnJIdXUa-JC9VQ1klHvjMJ4t0rQKyPqd_BgJ5beuyjBmpCXkE5uR-4KolLZKlwCzPll8TzpRvMSQiamx1c6MqQmP2Vy-gOGfi-LTNNZhcr4lmyEVc73okFZAAB0nbnZ-GLN7qRc0ZtBLjOdR-R60yhihCw_nEKP3auvKnzi7hC_DwdT8bxkbcHAEE4AeyvERUm6dleoMqax55qLO2sHQ8ZnnsP8eUFDGvDn--rLyNIa0jklNW5LWLVIF3gmmpTOyQ2wldElJqmllhQ',
    'W7Sb3oA56Cp_sdRLQ7UzUiPlGqOZr24UVzY7ZB84MuZHM-TnDgi9QJTE5fX87JM6Q-_TyDjI5DY0l1CCoHIohbjPJTZAhmfyJfbeMsQH7nZv0hGlzRIUA_5bpLjHQPQk_X5J-_9RvCe5wFUq9d1h3Uvz_UTNLoF2j26nENFdDJ2JTsJ0OH3ihsQ5LYPAcsNtTET0gFSJWfBClqfrFbPcGe1Y54sMvBuf8fNMhhemPDIP3HGTpbSY-OmQ-jcvLp30ld38LQEklBReI5cS5J9vpST0fgkZ02v-Ex-s1Yt6TErldBOpjEAhVCMTciFLepp1kzyNO0AhlfWXl7QoEsSfWQ',
    'CUdWRfjVbbDnZNLyuJjgiEr85ngGZ4DXjKDhGCKX5pH2NVkNwIwM1y9zw9FN2MiSFG8tzWdjCBxkVSmGzqhFCkz65Gx5OffnD57iHy0cc56bgGfNq-WHlqvAUiAJX3HzJ5MRn2HDZ4eGm0bmBQDhB91qvDpNIyIaLHgxWl8Brmn-tLuxIt4B1kpKKFy-crCeAsdHkZABFvGmtNeelH_XpJWc-5DKTd6JtpIhpPrsUKd8iFvWYJkc9Vp1k3OXTZARzuJ7qkt2xCBC38wKU4U_KXYmsJ5fAUuXWwk4AoirYlGalLLXIppg3T_Lg8AGKrJpM1tXD-GbZyK4tL1M2O2lSg',
    'jYW04zLDHfR1v7xdrW3lCGZrMIsVe0vWCfVkN2DRns2c3MN-mcp_-RE6TN9umSBYoNV-mnb31wFf8iun3fB6aDS6m_OXAiURVEKrPFNGlR38JSHUtsFzqTOj-wFrJZN4RwvZnNGSMvK3wzzUriZqmiNLsG8lktlEn6KA4kYVaM61_NpmPHWAjGExWv7cjHYupcjMSmR8uMTwN5UuAwgW6FRstCJEfoxwb0WKiyoaSlDuIiHZJ0cyGhhEmmAPiCwtPAwGeaL1yZMcp0p82cpTQ5Qb-7CtRov3N4DcOHgWYk6LomPR5j5cCkePAz87duqyzSMpCB0mCOuE3CU2VMtGeQ', // Sample kenjis
    'tyh-VfuzIxCyGYDlkBA7DfyjrqmSHu6pQ2hoZuFqUSLPNY2N0mpHb3nk5K17HWP_3cYHBw7AhHale5wky6-sVA', // ES256
    'VUPWQZuClnkFbaEKCsPy7CZVMh5wxbCSpaAWFLpnTe9J0--PzHNeTFNXCrVHysAa3eFbuzD8_bLSsgTKC8SzHxRVSj5eN86vBPo_1fNfE7SHTYhWowjY4E_wuiC13yoj', // ES384
    'AbVUinMiT3J_03je8WTOIl-VdggzvoFgnOsdouAs-DLOtQzau9valrq-S6pETyi9Q18HH-EuwX49Q7m3KC0GuNBJAc9Tksulgsdq8GqwIqZqDKmG7hNmDzaQG1Dpdezn2qzv-otf3ZZe-qNOXUMRImGekfQFIuH_MjD2e8RZyww6lbZk', // ES512
    'iOeNU4dAFFeBwNj6qdhdvm-IvDQrTa6R22lQVJVuWJxorJfeQww5Nwsra0PjaOYhAMj9jNMO5YLmud8U7iQ5gJK2zYyepeSuXhfSi8yjFZfRiSkelqSkU19I-Ja8aQBDbqXf2SAWA8mHF8VS3F08rgEaLCyv98fLLH4vSvsJGf6ueZSLKDVXz24rZRXGWtYYk_OYYTVgR1cg0BLCsuCvqZvHleImJKiWmtS0-CymMO4MMjCy_FIl6I56NqLE9C87tUVpo1mT-kbg5cHDD8I7MjCW5Iii5dethB4Vid3mZ6emKjVYgXrtkOQ-JyGMh6fnQxEFN1ft33GX2eRHluK9eg', // PS256
    'J5W09-rNx0pt5_HBiydR-vOluS6oD-RpYNa8PVWwMcBDQSXiw6-EPW8iSsalXPspGj3ouQjAnOP_4-zrlUUlvUIt2T79XyNeiKuooyIFvka3Y5NnGiOUBHWvWcWp4RcQFMBrZkHtJM23sB5D7Wxjx0-HFeNk-Y3UJgeJVhg5NaWXypLkC4y0ADrUBfGAxhvGdRdULZivfvzuVtv6AzW6NRuEE6DM9xpoWX_4here-yvLS2YPiBTZ8xbB3axdM99LhES-n52lVkiX5AWg2JJkEROZzLMpaacA_xlbUz_zbIaOaoqk8gB5oO7kI6sZej3QAdGigQy-hXiRnW_L98d4GQ', // PS512
);

foreach ($str as $key => $val){
    echo mb_strlen($val).PHP_EOL;
}
```